refactor/manager_page

DotsenkoS opened the pull request:
Кабинет менеджера приведен к виду кабинета Админа. 
Поправлен код в js в соответствующих функциях. 
Доступ для страниц менеджера (/manager/**", /api/manager/**) теперь есть только у менеджера
В нескольких местах поправлен формат ответа с просто статуса на ResponseDto
В ProductsRestController поправлена ошибка в методе filterByCategoryAndSort

DotsenkoS committed:
Кабинет менеджера приведен к виду кабинета Админа.
Поправлен код в js в соответствующих функциях.
Доступ для страниц менеджера (/manager/**", /api/manager/**) теперь есть только у менеджера
В нескольких местах поправлен формат ответа с просто статуса на ResponseDto
В ProductsRestController поправлена ошибка в методе filterByCategoryAndSort

DotsenkoS committed:
Merge branch 'dev' into refactor/manager_page

# Conflicts:
#	src/main/java/com/jm/online_store/controller/simple/ManagerController.java
#	src/main/resources/templates/manager-page.html

DotsenkoS committed:
Кабинет менеджера приведен к виду кабинета Админа.
Поправлен код в js в соответствующих функциях.
Доступ для страниц менеджера (/manager/**", /api/manager/**) теперь есть только у менеджера
В нескольких местах поправлен формат ответа с просто статуса на ResponseDto
В ProductsRestController поправлена ошибка в методе filterByCategoryAndSort

fponin commented on src/main/resources/templates/manager-page.html:
```diff
@@ -1,60 +1,2127 @@
+        </div>
+
+        <!-- ОСНОВНАЯ  ПАНЕЛЬ  -->
+        <div class="col-sm-8 text-left">
```
пока правой навигации нет, может 10 поставить(у админа 10)?
<div class="col-sm-10 text-left">

fponin reviewed (commented):
Все хорошо.

DotsenkoS committed:
Кабинет менеджера приведен к виду кабинета Админа.
Поправлен код в js в соответствующих функциях.
Доступ для страниц менеджера (/manager/**", /api/manager/**) теперь есть только у менеджера
В нескольких местах поправлен формат ответа с просто статуса на ResponseDto
В ProductsRestController поправлена ошибка в методе filterByCategoryAndSort
+Исправлено по замечаниям

fponin reviewed (commented):
Все отлично

tediik commented on src/main/java/com/jm/online_store/controller/rest/manager/ManagerTopicsCategoryRestController.java:
```diff
@@ -49,10 +49,9 @@
             @ApiResponse(code = 200, message = "TopicCategories haven't been found")
     })
-    public ResponseEntity<ResponseDto<List<TopicsCategoryDto>>> readAllTopicsCategories() {
+    public ResponseEntity<ResponseDto<List<TopicsCategory>>> readAllTopicsCategories() {
```
слушай, а тут точно надо было убирать TopicsCategoryDto? Проверь

DotsenkoS replied:
Да, тут точно - у нас в manager-categories-setting.js в функциях makeActualCategoryBody и makeArchiveCategoryBody нужны данные поля topics, которое есть в  TopicsCategory, но нет TopicsCategoryDto
Это было изменено, когда в классе ManagerTopicsCategoryRestController возвращаемое значение меняли на  ResponseDto. Тогда фронт не исправляли - возможно недопроверили что именно будет нужно на выходе, не могу сказать. В общем, с TopicsCategoryDto сейчас не будет работать, нужен именно TopicsCategoryДа, тут точно - у нас в manager-categories-setting.js в функциях makeActualCategoryBody и makeArchiveCategoryBody нужны данные поля topics, которое есть в  TopicsCategory, но нет TopicsCategoryDto
Это было изменено, когда в классе ManagerTopicsCategoryRestController возвращаемое значение меняли на  ResponseDto. Тогда фронт не исправляли - возможно недопроверили что именно будет нужно на выходе, не могу сказать. В общем, с TopicsCategoryDto сейчас не будет работать, нужен именно TopicsCategory

tediik commented on src/main/java/com/jm/online_store/controller/simple/ManagerController.java:
```diff
@@ -20,66 +20,10 @@
         User user = userService.getCurrentLoggedInUser();
         model.addAttribute("user", user);
-        return "manager-profile";
+        model.addAttribute("listCategories", categoriesService.getCategoriesWithoutParentCategory().stream().map(Categories::getCategory).collect(Collectors.toList()));
```
трабла в том, что это никак не увидят фронты, эту инфу должен отдавать рест-эндпоинт и мы должны получать ее через ajax-запрос, т.е. если ты берешься переделать, то тогда надо делать сразу норм

DotsenkoS replied:
Исправила

tediik commented on src/main/resources/templates/manager-page.html:
```diff
@@ -1,60 +1,2123 @@
+                        </div>
+                    </div>
+                </div>
+                <!--ВКЛАДКА ОТЧЕТЫ   -->
+                <!--ВКЛАДКА РАССЫЛКИ  -->
```
Я понимаю, что там везде сделано так же, как думаешь, может стоило вынести вкладки в отдельные фрагменты, чтобы просто было проще вносить правки в файл, не?

DotsenkoS replied:
Под конец это стало настоящей дилеммой, до сих пор не знаю как было бы лучше

DotsenkoS committed:
Заполнение спиков категорий товаров теперь осуществляется через js

DotsenkoS committed:
Заполнение спиков категорий товаров и профиль менеджера теперь осуществляется через js
